runtime_scan/pkg/orchestrator/orchestrator.go

```diff
@@ -85,12 +85,13 @@ func (o *Orchestrator) Stop() {
 	o.server.Stop()
 }
 
-func (o *Orchestrator) Scan(scanConfig *_config.ScanConfig, scanDone chan struct{}) error {
-	if err := o.getScanner().Scan(scanConfig, scanDone); err != nil {
-		return fmt.Errorf("failed to scan: %v", err)
+func (o *Orchestrator) Scan(scanConfig *_config.ScanConfig) (chan struct{}, error) {
+	o.scanner = _scanner.CreateScanner(o.config, o.clientset)
+	done, err := o.getScanner().Scan(scanConfig)
+	if err != nil {
+		return done, fmt.Errorf("failed to scan: %v", err)
```

Comment:
maybe better to return nil here instead of done?
Not sure it's clear what to do when you get both an error and a channel

Reply:
updated it so that it returns nil whenever there is an error to be returned and moved the channel setup later in the functions that return an error:
https://github.com/openclarity/kubeclarity/compare/750c55cd399c68f262348dd7dca6b5c51d289f2f..994d85defae90bcfd22cb5c165ada99ddfa8fd63